### DISCUSSION TITLE
Using user input variables as keys

### DISCUSSION THREAD

**Discussion by CarmeloDascola** [prisma/prisma#10028] [Q&A]
Hello, I have a schema.prisma that looks like this:
```prisma
model GuildSettings {
	@@map("guild")

	// Basics
	guildID          String   @id @map("guild_id")
	prefix           String?  @default("-")
	language         String   @default("en-US")
}
```
If I want for instance to modify the language I can do:
```ts
await this.container.client.prisma.guildSettings.update({
	where: { guildID: message.guild?.id },
	data: { language: language },
})
```
What if I want the field (e.g. prefix or language) to be an user input?
If I want to set the key as a variable it gives me this:
```
Type '{ myVariable: string | Prisma.StringFieldUpdateOperationsInput; }' is not assignable to type '(Without<GuildSettingsUpdateInput, GuildSettingsUncheckedUpdateInput> & GuildSettingsUncheckedUpdateInput) | (Without<...> & GuildSettingsUpdateInput)'.
  Object literal may only specify known properties, and 'cmd' does not exist in type '(Without<GuildSettingsUpdateInput, GuildSettingsUncheckedUpdateInput> & GuildSettingsUncheckedUpdateInput) | (Without<...> & GuildSettingsUpdateInput)'.ts(2322)
```
How can I do it?
Thanks...

**Comment by austincrim** [comment]
Hey @CarmeloDascola 👋, thanks for the question.

The use case you're describing should be possible! I am able to run this code snippet without any issues:
```prisma
// schema.prisma
model Post {
  id      Int @id @default(autoincrement())
  content String
}
```
```ts
const key = 'content'
await prisma.post.create({
  data: {
    [key]: 'Hello world!'
   }
})
```

Could you share the code that is causing the type error you posted above?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
